src/flags.zig

```diff
@@ -296,101 +296,135 @@ fn parse_value(comptime T: type, flag: []const u8, value: [:0]const u8) T {
-            .bytes = value,
-            .original = "error", // this shouldn't ever be visible in the UI
+            .value = value,
+            .unit = ByteUnit.bytes,
         };
     }
-};
 
-fn parse_value_size(flag: []const u8, value: []const u8) ByteSize {
-    assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+    fn parse(flag: []const u8, value: []const u8) ByteSize {
+        assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
 
-    const units = .{
-        .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, 1024 * 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, 1024 * 1024 * 1024 },
-        .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, 1024 * 1024 },
-        .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, 1024 },
-    };
+        const units = .{
+            .{ &[_][]const u8{ "TiB", "tib", "TB", "tb" }, ByteUnit.tb },
+            .{ &[_][]const u8{ "GiB", "gib", "GB", "gb" }, ByteUnit.gb },
+            .{ &[_][]const u8{ "MiB", "mib", "MB", "mb" }, ByteUnit.mb },
+            .{ &[_][]const u8{ "KiB", "kib", "KB", "kb" }, ByteUnit.kb },
+        };
 
-    const unit: struct { suffix: []const u8, scale: u64 } = unit: inline for (units) |unit| {
-        const suffixes = unit[0];
-        const scale = unit[1];
-        for (suffixes) |suffix| {
-            if (std.mem.endsWith(u8, value, suffix)) {
-                break :unit .{ .suffix = suffix, .scale = scale };
+        const unit: struct {
+            suffix: []const u8,
+            unit: ByteUnit,
+        } = unit: inline for (units) |unit| {
+            const suffixes = unit[0];
+            const unitkind = unit[1];
+            for (suffixes) |suffix_| {
+                if (std.mem.endsWith(u8, value, suffix_)) {
+                    break :unit .{ .suffix = suffix_, .unit = unitkind };
+                }
             }
-        }
-    } else break :unit .{ .suffix = "", .scale = 1 };
+        } else break :unit .{ .suffix = "", .unit = ByteUnit.bytes };
+
+        assert(std.mem.endsWith(u8, value, unit.suffix));
+        const value_numeric = value[0 .. value.len - unit.suffix.len];
+
+        const amount = std.fmt.parseUnsigned(u64, value_numeric, 10) catch |err| {
+            switch (err) {
+                error.Overflow => fatal(
+                    "{s}: value exceeds 64-bit unsigned integer: '{s}'",
+                    .{ flag, value },
+                ),
+                error.InvalidCharacter => fatal(
+                    "{s}: expected a size, but found '{s}' (invalid digit or suffix)",
+                    .{ flag, value },
+                ),
+            }
+        };
 
-    assert(std.mem.endsWith(u8, value, unit.suffix));
-    const value_numeric = value[0 .. value.len - unit.suffix.len];
+        _ = std.math.mul(u64, amount, @intFromEnum(unit.unit)) catch fatal(
+            "{s}: size in bytes exceeds 64-bit unsigned integer: '{s}'",
+            .{ flag, value },
+        );
```

Comment:
Is there a prettier way to write this overflow check that discards the actual result?

Reply:
The built-in `@mulWithOverflow` returns the result and the overflow bit. The function `std.math.mul()` is just a wrapper to make it idiomatic with errors;

I think we need to discard the result anyway!